app/services/link_reporter_csv_service.rb

```diff
@@ -0,0 +1,66 @@
+class LinkReporterCsvService
+  def initialize(reports_dir:, organisation:)
+    @organisation = organisation
+    @reports_dir = reports_dir
+  end
+
+  def generate
+    CSV.open(file_path, "w", encoding: "UTF-8") do |csv|
+      csv << headings
+      public_editions.each { |edition| csv << row_for_edition(edition) }
+    end
+  end
+
+private
+
+  attr_reader :organisation, :reports_dir
+
+  def public_editions
+    Edition.publicly_visible.with_translations.in_organisation(organisation)
+  end
+
+  def file_path
+    reports_dir.join("#{@organisation.slug}_links_report.csv")
+  end
+
+  def headings
+    ["page", "admin link", "public timestamp", "format", "broken link count", "broken links"]
+  end
+
+  def row_for_edition(edition)
+    [
+      public_url(edition),
+      admin_url(edition),
+      timestamp(edition),
+      edition.type,
+      broken_links(edition).size,
+      broken_links(edition).join("\r\n")
```

Comment:
Does having new lines here work with the CSV output?

Reply:
Yup, and to make sure the new CSV remained consistent with the old ones I had copied this from here: https://github.com/alphagov/whitehall/blob/master/lib/whitehall/broken_link_reporter.rb#L44